hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/SparkHoodieTableFileIndex.scala

```diff
@@ -417,7 +417,7 @@ class SparkHoodieTableFileIndex(spark: SparkSession,
       schema,
       metaClient.getTableConfig.propsMap,
       configProperties.getString(DateTimeUtils.TIMEZONE_OPTION, SQLConf.get.sessionLocalTimeZone),
-      sparkParsePartitionUtil,
+      SparkBaseParsePartitionUtil,
```

Comment:
If `SparkBaseParsePartitionUtil` is all the same across Spark versions, there is no need to pass this object around.